tests/common/test_os_calls.c

```diff
@@ -0,0 +1,51 @@
+
+#if defined(HAVE_CONFIG_H)
+#include "config_ac.h"
+#endif
+
+#include <stdlib.h>
+#include "test_common.h"
+#include "os_calls.h"
+
+START_TEST(test_g_file_get_size__returns_file_size)
+{
+    unsigned long long size;
+
+    size = g_file_get_size("../../xrdp/xrdp256.bmp");
+    ck_assert_int_eq(size, 49278);
+
+
+    size = g_file_get_size("../../xrdp/ad256.bmp");
+    ck_assert_int_eq(size, 19766);
+
+}
+END_TEST
+
+START_TEST(test_g_file_get_size__more_than_4GiB)
+{
+    unsigned long long size;
+
+    system("dd if=/dev/zero of=./file_5GiB.dat bs=4k count=1310720");
+
+    size = g_file_get_size("./file_5GiB.dat");
+    ck_assert_int_eq(size, 5368709120);
```

Comment:
This fails as assumed until `g_file_get_size()` gets fixed to be able to handle more than 4GiB files.